### ISSUE TITLE
High memory utilization after router upgrade to 0.91.0

### ISSUE THREAD

**Issue by PeteMac88** [wundergraph/cosmo#854]
### Component(s)

router

### Component version

0.91.0

### wgc version

latest

### controlplane version

latest

### router version

0.91.0

### What happened?

After upgrading from router version 0.84.4 we observe much higher memory utilization of the router instances. Initially after the upgrade all instances were OOMKilled because the pod hit the memory limits. We are running the router in a k8s cluster so here are the resources before the update which were fine for the older router version:

```
resources:
   requests:
        cpu: 400m
        memory: 256Mi
   limits:
        cpu: 1000m
        memory: 600Mi
```

Now memory utilization increased to around 1000Mi. Do you have an explanation for this? 

## Environment
Kubernetes 1.25

**Comment by jensneuse** [comment]
We're taking a look

**Comment by devsergiy** [comment]
Hi @PeteMac88 
Could you check with the `0.90.0` version?

**Comment by PeteMac88** [comment]
Hey @devsergiy, es definitely higher as the  0.84.4 version but not as high as 0.91.0. Here is a table with the average utilization with same load:

0.84.4 ~ 350Mi
0.90.0 ~ 700 Mi
0.91.0. ~ 1200Mi

**Comment by devsergiy** [comment]
Thanks a lot

Could you also check `0.88.0`?

We already know the reasons why it changes in 0.90.1
But it is a bit of a surprise that it also spikes before that on 0.90.0

**Comment by PeteMac88** [comment]
@devsergiy Any update here? I needed to downgrade the router yesterday because of the increasing memory consumption.

**Comment by Aenimus** [comment]
Moreover, were you able to check 0.88.0 as requested @PeteMac88?

**Comment by jensneuse** [comment]
Are you able to build the router from source like so:
```bash
go build -tags=pprof main.go
```

Then run the router with your regular traffic and run the following command:

```bash
go tool pprof http://localhost:6060/debug/pprof/heap
```

You need to have the go toolchain installed for this.

Once you've got a heap profile, can you upload it here for investigation? Thanks

**Comment by jfroundjian** [comment]
I also ran into this problem when upgrading to 0.95.6 this week. I observed unbounded memory growth when load testing the router with our batch requests > 100 list items and ~ 150 fields across 2 subgraphs (No deep nesting & max 2 levels).

I did not run into this problem with the same queries that only returned a single item. 

Reverting to 0.88.0 definitely helpled and the containers use less memory now.
